go/pkg/client/cas.go

```diff
@@ -581,6 +581,29 @@ func (c *Client) FlattenActionOutputs(ctx context.Context, ar *repb.ActionResult
 	return outs, nil
 }
 
+// DownloadDirectory downloads the entire directory of given digest.
+func (c *Client) DownloadDirectory(ctx context.Context, d digest.Digest, execRoot string, cache filemetadata.Cache) error {
+	dir := &repb.Directory{}
+	if err := c.ReadProto(ctx, d, dir); err != nil {
+		return err
```

Comment:
It will useful to have more informative error description here:
```
return fmt.Errorf("digest %v cannot be mapped to a directory proto: %v", d, err)
```

Reply:
changed.